test/python/quantum_info/test_qubit_sparse_pauli.py

```diff
@@ -0,0 +1,989 @@
+
+    def test_empty(self):
+        empty_5 = QubitSparsePauliList.empty(5)
+        self.assertEqual(empty_5.num_qubits, 5)
+        self.assertEqual(len(empty_5), 0)
+        self.assertEqual(empty_5.to_sparse_list(), [])
+
+        empty_0 = QubitSparsePauliList.empty(0)
+        self.assertEqual(empty_0.num_qubits, 0)
+        self.assertEqual(len(empty_0), 0)
+        self.assertEqual(empty_0.to_sparse_list(), [])
+
+    def test_len(self):
+        self.assertEqual(len(QubitSparsePauliList.empty(0)), 0)
+        self.assertEqual(len(QubitSparsePauliList.empty(10)), 0)
+        self.assertEqual(len(QubitSparsePauliList.from_list(["IIIXIZ", "YYXXII"])), 2)
+
+    def test_pauli_enum(self):
+        # These are very explicit tests that effectively just duplicate magic numbers, but the point
+        # is that those magic numbers are required to be constant as their values are part of the
+        # public interface.
+
+        self.assertEqual(
+            set(QubitSparsePauli.Pauli),
+            {
+                QubitSparsePauli.Pauli.X,
+                QubitSparsePauli.Pauli.Y,
+                QubitSparsePauli.Pauli.Z,
+            },
+        )
+        # All the enumeration items should also be integers.
+        self.assertIsInstance(QubitSparsePauli.Pauli.X, int)
+        values = {
+            "X": 0b10,
+            "Y": 0b11,
+            "Z": 0b01,
+        }
+        self.assertEqual({name: getattr(QubitSparsePauli.Pauli, name) for name in values}, values)
+
+        # The single-character label aliases can be accessed with index notation.
+        labels = {
+            "X": QubitSparsePauli.Pauli.X,
+            "Y": QubitSparsePauli.Pauli.Y,
+            "Z": QubitSparsePauli.Pauli.Z,
+        }
+        self.assertEqual({label: QubitSparsePauli.Pauli[label] for label in labels}, labels)
+        # The `label` property returns known values.
+        self.assertEqual({pauli.label: pauli for pauli in QubitSparsePauli.Pauli}, labels)
+
+    @ddt.idata(single_cases_list())
+    def test_pickle(self, qubit_sparse_pauli_list):
+        self.assertEqual(qubit_sparse_pauli_list, copy.copy(qubit_sparse_pauli_list))
+        self.assertIsNot(qubit_sparse_pauli_list, copy.copy(qubit_sparse_pauli_list))
+        self.assertEqual(qubit_sparse_pauli_list, copy.deepcopy(qubit_sparse_pauli_list))
+        self.assertEqual(
+            qubit_sparse_pauli_list, pickle.loads(pickle.dumps(qubit_sparse_pauli_list))
+        )
+
+    @ddt.data(
+        # This is every combination of (0, 1, many) for (terms, qubits, non-identites per term).
+        QubitSparsePauliList.empty(0),
+        QubitSparsePauliList.empty(1),
+        QubitSparsePauliList.empty(10),
+        QubitSparsePauliList.from_label("IIXIZI"),
+        QubitSparsePauliList.from_label("X"),
+        QubitSparsePauliList.from_list(["YIXZII"]),
+        QubitSparsePauliList.from_list(["YIXZII", "ZZYYXX"]),
+    )
```

Comment:
Not sure the comment is accurate here either, but then again, I wouldn't 100% swear I was correct in my original either.

Reply:
Likewise dropped the comment here, and added the test case:
```
QubitSparsePauliList.from_list(["IIIIII", "ZZYYXX"])
```
(adding a case with an identity operator).